app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -714,10 +712,12 @@ class HomeFragment : Fragment() {
             // We want to show the popup only after privateBrowsingButton is available.
             // Otherwise, we will encounter an activity token error.
             privateBrowsingButton.post {
```

Comment:
Can you help me make this reproducible? I've tried the STR in https://github.com/mozilla-mobile/fenix/issues/17177#issue-772508613 but no luck.

For me `showCondition` makes sure the popup is not displayed with the STR above: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/utils/Settings.kt#L856

`numTimesPrivateModeOpened.value` is only incremented when tapping the private browsing mode button on Home, but not when opening via the FAB. On a clean build, `numTimesPrivateModeOpened.value` is 0 and the popup should never be displayed. Once we hit the button on home and the popup is finally displayed we toggle off `showedPrivateModeContextualFeatureRecommender` and shouldn't show it again either.

There must be something else going on here I am missing? If so let's add a comment to the code here as well :).

Reply:
you're right I can't repro with the original STR I wrote... 

I think I can still repro by getting to home screen from the FAB and then toggling the PBM button

![ezgif com-video-to-gif(2)](https://user-images.githubusercontent.com/25442310/103579360-08cdea80-4e8d-11eb-876a-3f9c0333a79a.gif)

Reply:
OK, got it. Let's add a comment above the `!bundleArgs.getBoolean(FOCUS_ON_ADDRESS_BAR)` check to describe why we're doing it? I think it will help in the future :).